lib/minitest/reporters/base_reporter.rb

```diff
@@ -85,12 +85,13 @@ def print(*args)
         io.print(*args)
       end
 
-      def print_info(e, name=true)
-        print "#{e.exception.class.to_s}: " if name
+      def print_info(e, error=true)
         e.message.each_line { |line| print_with_info_padding(line) }
 
-        trace = filter_backtrace(e.backtrace)
-        trace.each { |line| print_with_info_padding(line) }
+        unless error
+          trace = filter_backtrace(e.backtrace)
+          trace.each { |line| print_with_info_padding(line) }
+        end
```

Comment:
So, the issue is mainly just that `print_info` doesn't need to print backtrace when `e` is an error (which is an instance of `UnexpectedError`), because `UnexpectedError#message` is already much [verbose](https://github.com/seattlerb/minitest/blob/e9752484d624950bb3af4e96fde118d921002dd2/lib/minitest.rb#L699-L702) (contains the original error name, message and backtrace).

It looks like `print_info` is already used (at least by inbuilt reporters) such that the second param is set to `false` when `e` is an assertion, so this should be okay? (hence also renamed that boolean to `error`).